packages/core/src/RenderingEngine/WSIViewport.ts

```diff
@@ -611,12 +653,15 @@ class WSIViewport extends Viewport {
   protected canvasToIndex = (canvasPos: Point2): Point2 => {
     const transform = this.getTransform();
     transform.invert();
-    return transform.transformPoint(
+    const indexPoint = transform.transformPoint(
       canvasPos.map((it) => it * devicePixelRatio) as Point2
     );
+    indexPoint[1] = indexPoint[1] * -1; // Reverse y axis to match CS3D
```

Comment:
This is probably ok for now, but how you setup the inverse actually depends on the rotation/vectors in use.